src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -92,10 +93,69 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
           handler.destroy();
         });
 
+        function hasQueryOrFilters(searchSource) {
+          const filters = searchSource.get('filter');
+          if (filters.length > 0) {
+            return true;
+          }
+
+          const query = searchSource.get('query');
+          if (query.query && query.query.length > 0) {
+            return true;
+          }
+
+          return false;
+        }
+
+        function getFilterLabel(filter) {
+          if (!_.has(filter, 'meta') || (!filter.meta.alias &&
+            (!filter.meta.key && !filter.meta.value)) ) {
+            return '';
+          }
+
+          let prefix = _.get(filter, 'meta.negate', false) ? 'NOT' : '';
+          if (filter.meta.alias) {
+            return `"${prefix} ${filter.meta.alias}"`
+          }
+
+          return `"${prefix} ${filter.meta.key}:${filter.meta.value}"`;
+        }
+
+        function getSearchSourceLabel(searchSource) {
+          let label = '';
+          const filters = searchSource.get('filter');
+          if (filters.length > 0) {
+            const filterLabels = filters.map(filter => {
+              return getFilterLabel(filter);
+            });
+            label = `filters=${filterLabels.join(',')}`;
+          }
+
+          const query = searchSource.get('query');
+          if (query.query && query.query.length > 0) {
+            label = `${label} query="${query.query}"`;
+          }
+
+          return label;
+        }
+
+        let hasSearch = false;
+        let searchLabel = '';
+        if (hasQueryOrFilters(savedObject.searchSource)) {
+          hasSearch = true;
+          searchLabel = `This visualization has the following search criteria applied: ${getSearchSourceLabel(savedObject.searchSource)}`;
+        }
+        if (savedObject.savedSearchId) {
+          hasSearch = true;
+          const savedSearch = await this.searchLoader.get(savedObject.savedSearchId)
+          searchLabel = `${searchLabel} This visualization is linked to Saved Search "${savedSearch.title}" that applies the following search criteria: ${getSearchSourceLabel(savedSearch.searchSource)}`;
```

Comment:
One concern I have is that in dashboards (and Kibana generally) loading is already pretty chatty and slow. How often is it likely that this conditional will be true, do you think? Is there some way to maybe go ahead and render the embeddable / dashboard, and only load this stuff up when the tooltip is requested?